scripts/link.js

```diff
@@ -0,0 +1,51 @@
+const path = require('path');
+const fs = require('fs');
+const { execSync } = require('child_process');
+
+const COMPONENTS_PATH = path.join(__dirname, '../components');
+
+function getPackageJSON(component) {
+    const packageJSONPath = path.join(COMPONENTS_PATH, component, 'package.json');
+    if (!fs.lstatSync(packageJSONPath).isFile()) return null;
+    return JSON.parse(fs.readFileSync(packageJSONPath));
+}
+
+function main() {
+    // link the components library
+    execSync('npm link', { cwd: path.join(__dirname, '../lib') });
+    const components = fs.readdirSync(COMPONENTS_PATH);
+
+    // loop all components once to create links to the global node_modules.
+    // This is needed because later we'll link a component's local dependencies
+    // to the global node modules and we need to make sure all components already have links;
+    // otherwise we would have to recursively go through all dependencies, which is slow
+    // and error prone. NPM works like this, but we have an advantage in knowing the list
+    // of all dependencies beforehand - everything in the /components folder and the /lib folder.
+    for (let component of components) {
+        const links = execSync('npm ls -g --depth=0 --link=true', {encoding: 'utf8'});

```

Comment:
Turns out the errors is caused by missing dependencies that failed to get installed because of the "broken" uncommitted component. 
Tried to workaround it - couldn't.
Can't check for existing links as npm ls is what fails :D
The NODE_PATH variable doesn't exist on my PC so I can't rely on it to get the global node_modules folder and delete only the coherent-gameface-* folders.

Ended up always making the links despite of the build time cost - build:dev is slow, rebuild and build are less slow.

Reply:
Also added unlink script.